cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/documentstore/Attachment.java

```diff
@@ -37,10 +37,11 @@
  */
 public abstract class Attachment implements Comparable<Attachment>{
 
-    public Attachment(String name, String type, Encoding encoding) {
+    public Attachment(String name, String type, Encoding encoding, long length) {
```

Comment:
Given that this class is `abstract` should this constructor really be `protected` rather than `public`?

Since we only document `public` visibility I guess the question really is, should we be documenting this constructor so that people have guidance if they are constructing their own subclasses or do we consider our subclasses sufficient in which case we should mark this protected and omit it from the docs?